dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/data/NeuralNetworkDataParser.java

```diff
@@ -81,11 +79,10 @@ public void parse() {
         continue;
       }
       try {
-        final INDArray input = readNumpy(
-            new ByteArrayInputStream(text.getBytes()), delimiter);
-        final INDArray data = input.get(NDArrayIndex.interval(0, input.columns() - 2)).reshape(inputShape);
-        final int label = (int) input.getFloat(input.columns() - 2);
-        final boolean isValidation = ((int) input.getFloat(input.columns() - 1) == 1);
+        final Matrix input = readNumpy(matrixFactory, new ByteArrayInputStream(text.getBytes()), delimiter);
+        final Matrix data = input.get(range(0, input.getColumns() - 2));
```

Comment:
According to the comments on `range()`, the end index is exclusive. Shouldn't the second argument be `input.getColumns() - 1`?